Kianoosh76/IBS
# [IBS-70] Refactored submission endpoints for staff endpoint

## Description

- jira: https://ibs-utm.atlassian.net/browse/IBS-70
- Refactored submission endpoints for staff, and updated related files and models.

## Changes Made

- Set `token_count` for group_user to 10 instead for better maintain the test case performance.

- Set `timestamps` to false for submission to avoid unnecessary columns being asked in the query such as `createdAt` and `updatedAt`


## Test Guidelines

1. In `/backend/` run the following and ensure that they all passes:
 ```bash 
npm test submission_staff_check
```
<img width="588" alt="Screenshot 2024-05-01 at 2 40 01 AM" src="https://github.com/Kianoosh76/IBS/assets/71187784/11b46c17-388e-487c-b999-a6e2761a7a95">

 ```bash 
npm test submission_staff_collect_all
```
<img width="619" alt="Screenshot 2024-05-01 at 2 40 27 AM" src="https://github.com/Kianoosh76/IBS/assets/71187784/6cd94c90-1826-4135-aed3-985c9e69afc6">

 ```bash 
npm test submission_staff_collect_manual
```
<img width="545" alt="Screenshot 2024-05-01 at 2 40 42 AM" src="https://github.com/Kianoosh76/IBS/assets/71187784/66c78081-dee7-4293-b5b7-f62c847e104a">

 ```bash 
npm test submission_staff_collect_one
```
<img width="647" alt="Screenshot 2024-05-01 at 2 41 04 AM" src="https://github.com/Kianoosh76/IBS/assets/71187784/229912c1-5eb5-4465-8f8b-f0611ccaeac4">


 ```bash 
npm test submission_staff_download
```
<img width="584" alt="Screenshot 2024-05-01 at 2 41 23 AM" src="https://github.com/Kianoosh76/IBS/assets/71187784/b769b309-7261-411b-8e31-8783979ad383">


## Checklist

### For PR author:
- [ ] I have added appropriate comments to the code.
- [ ] I have ensured code style and formatting are consistent.
- [ ] I have included unit tests or integration tests where applicable.
- [ ] I have considered edge cases and handled them appropriately.
- [ ] I have celebrated my accomplishments 🎉

### For PR reviewer:
- [ ] I have reviewed the code changes.
- [ ] I have verified that comments are clear and helpful.
- [ ] I have ensured code style and formatting align with project standards.
- [ ] I have examined any included tests and their coverage.
- [ ] I have considered potential edge cases and provided feedback on them.
